tests/mocha/insertion_marker_manager_test.js

```diff
@@ -0,0 +1,329 @@
+                            'block': {
+                              'type': 'stack_block',
+                              'id': 'third',
+                            },
+                          },
+                        },
+                      },
+                    },
+                ],
+                },
+            };
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 2);
+        });
+
+        test('One value block', function() {
+            const state = {
+              'blocks': {
+                'blocks': [
+                  {
+                    'type': 'row_block',
+                    'id': 'first',
+                  },
+                ],
+              },
+            };
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+        });
+
+        test('Two value blocks', function() {
+            const state = {
+              'blocks': {
+                'blocks': [
+                  {
+                    'type': 'row_block',
+                    'id': 'first',
+                    'inputs': {
+                        'INPUT': {
+                          'block': {
+                            'type': 'row_block',
+                            'id': 'second',
+                          },
+                        },
+                      },
+                  },
+                ],
+              },
+            };
+            const manager = createBlocksAndManager(this.workspace, state);
+            const markers = manager.getInsertionMarkers();
+            chai.assert.equal(markers.length, 1);
+        });
+    });
+
+    suite('Would delete block', function() {
+        setup(function() {
+            const state = {
+                'blocks': {
+                  'blocks': [
+                    {
+                      'type': 'stack_block',
+                      'id': 'first',
+                    },
+                  ],
+                },
+              };
+            Blockly.serialization.workspaces.load(state, this.workspace);
+            this.block = this.workspace.getBlockById('first');
+            this.manager = new Blockly.InsertionMarkerManager(this.block);
+        });
+        
+        function stubComponentManager(workspace) {
+            const componentManager = workspace.getComponentManager();
+            const stub = sinon.stub(componentManager, 'hasCapability');
+            return stub;
+        }
+
+        test('Over delete area: accepted', function() {
+            const dxy = new Blockly.utils.Coordinate(0, 0);
+            const stub = stubComponentManager(this.workspace);
+            stub.withArgs('fakeDragTarget',
+                Blockly.ComponentManager.Capability.DELETE_AREA).returns(true);
```

Comment:
nit: break at a higher syntactic level if possible. https://google.github.io/styleguide/jsguide.html#formatting-where-to-break

(Fixing the indentation may make this easier)